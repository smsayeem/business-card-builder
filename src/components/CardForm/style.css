
.form-container{
  display: flex;
  flex-direction: column;
  padding: 3rem 1rem 1rem 1rem;
  background-color: var(--mainWhite);
  width: 100%;
}

header h1{
  letter-spacing: 1px;
  font-size: 2rem;
}

.divider{
  border-bottom: 1px solid var(--mainGrey);
  margin: 2rem 0 1rem;
}
.divider p{
  text-transform: uppercase;
  padding-bottom: 6px;
  font-size: 0.8rem;
  color: var(--primaryLight);
}

.form-center {
  display: flex;
  flex-wrap: wrap;
  flex: 0 0 100%;
}

.form-group {
  flex: 0 0 100%;
}

label {
  display: block;
  color: var(--primaryGrey);
  font-size: 0.8rem;
  text-transform: capitalize;
  margin: 1rem 0 0.2rem;
  text-transform: uppercase;
}

.form-input {
  border: 1px solid var(--primaryLight);
  border-radius: 4px;
  outline: none;
  width: 100%;
  font-size: 1rem;
  padding: 10px;
  color: var(--mainBlack);
  letter-spacing: 1px;
}

.btn-group {
  padding: 1rem 0;
  margin: 1rem 0 0 0;
  width: 100%;
  display: flex;
  justify-content: space-between;
}

.btn {
  width: 48%;
  height: 2.7rem;
  text-decoration: none;
  color: var(--mainWhite);
  background-color: var(--mainBlack);
  text-align: center;
  text-transform: capitalize;
  letter-spacing: 1px;
  font-size: 1rem;
  outline: 0;
  border: none;
  border-radius: 4px;
  padding: 0 1rem;
  cursor: pointer;
  box-shadow: 0 2px 2px rgba(0, 0, 0, 0.2);
  transition: all 0.3s ease;
  -webkit-transition: all 0.3s ease;
  -moz-transition: all 0.3s ease;
  -ms-transition: all 0.3s ease;
  -o-transition: all 0.3s ease;
}
.btn:hover { box-shadow: 0 3px 3px rgba(0, 0, 0, 0.3); }
.btn-avatar{ background-color: var(--secondaryBtn); }
.btn-avatar:hover{ background-color: var(--secondaryBtnDark); }
.btn-hcard{ background-color: var(--primaryBtn); }
.btn-hcard:hover{ background-color: var(--primaryBtnDark); }

@media screen and (min-width: 430px){
  .form-container{ padding: 3rem; }
  .form-group { flex: 0 0 50%; }
  .form-group:nth-child(odd) { padding-right: 8px; }
  .form-group:nth-child(even) { padding-left: 8px; }
  .btn-group { margin: 1rem 0 0 0; }
  .btn{ font-size: 1.1rem; }
}

@media screen and (min-width: 769px){
  .form-container{ flex: 0 0 50%; }
}
