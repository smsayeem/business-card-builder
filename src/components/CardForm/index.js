import React from 'react';
import PropTypes from 'prop-types';

import './style.css';

const CardForm = ({ handleChange, handleSubmit, handleImage }) => {


  // when 'Upload Avatar' button is clicked this function targets the input ('#upload-button') and add a callback function to upload image
  const clickUpload=()=>{
    const uploadBtn = document.querySelector('#upload-button');
    uploadBtn.click();
    uploadBtn.addEventListener('change', handleImage)
  }

  return (
    <div className="form-container">
      <header>
        <h1>hCard Builder</h1>
      </header>
      <form onSubmit={handleSubmit}>
        <div className="divider">
          <p>personal details</p>
        </div>
        {/* input fields */}
        <section className="form-center">
          <div className="form-group">
            <label htmlFor="givenname">given name</label>
            <input 
              type="text" 
              id='givenname' 
              name='givenname' 
              className="form-input" 
              onChange={handleChange} />
          </div>
          <div className="form-group">
            <label htmlFor="surname">surname</label>
            <input 
              type="text" 
              id='surname' 
              name='surname' 
              className="form-input"           
              onChange={handleChange} />
          </div>
          <div className="form-group">
            <label htmlFor="email">email</label>
            <input 
              type="email" 
              id='email' 
              name='email' 
              className="form-input"           
              onChange={handleChange} />
          </div>
          <div className="form-group">
            <label htmlFor="phone">phone</label>
            <input 
              type="text" 
              id='phone' 
              name='phone' 
              className="form-input" 
              onChange={handleChange} />
          </div>
          </section>

          <div className="divider">
            <p>personal details</p>
          </div>
          <section className="form-center">
          <div className="form-group">
            <label htmlFor="houseno">house name or #</label>
            <input 
              type="text" 
              id='houseno' 
              name='houseno' 
              className="form-input" 
              onChange={handleChange} />
          </div>
          <div className="form-group">
            <label htmlFor="street">street</label>
            <input 
              type="text" 
              id='street' 
              name='street' 
              className="form-input" 
              onChange={handleChange} />
          </div>
          <div className="form-group">
            <label htmlFor="suburb">suburb</label>
            <input 
              type="text" 
              id='suburb' 
              name='suburb' 
              className="form-input" 
              onChange={handleChange} />
          </div>
          <div className="form-group">
            <label htmlFor="state">state</label>
            <input 
              type="text" 
              id='state' 
              name='state' 
              className="form-input" 
              onChange={handleChange} />
          </div>
          <div className="form-group">
            <label htmlFor="postcode">post code</label>
            <input 
              type="number" 
              id='postcode' 
              name='postcode' 
              className="form-input" 
              onChange={handleChange} />
          </div>
          <div className="form-group">
            <label htmlFor="country">country</label>
            <input 
              type="text" 
              id='country' 
              name='country' 
              className="form-input" 
              onChange={handleChange} />
          </div>
          {/* buttons */}
          <div className="btn-group ">
            {/* this input allows to select image files. It is active but hidden as per design  */}
            <input id="upload-button" type='file' hidden/>
            {/* this button will call a function 'clickUpload' that will trigger the input above to upload an image */}
            <button name='upload' className = "btn btn-avatar" onClick = {clickUpload}>Upload Avatar</button>
            {/* no functionality added for this button as it is not part of the test */}
            <button type="submit" name="submit" className = "btn btn-hcard" onClick={(e) => handleSubmit(e)}>
              Create hCard
            </button>
          </div>
        </section>
      </form>
    </div>
  )
}

CardForm.propTypes = {
  handleChange: PropTypes.func, 
  handleSubmit: PropTypes.func, 
  handleImage: PropTypes.func,
};

CardForm.defaultProps = {
  handleChange: ()=>{}, 
  handleSubmit: ()=>{}, 
  handleImage: ()=>{},
};

export default CardForm
