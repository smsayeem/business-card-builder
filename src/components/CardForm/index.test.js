import { shallow } from 'enzyme';
import React from 'react';
import CardForm from './index';

describe('Login Component', () => {
  let wrapper;
  beforeEach(()=>{
    wrapper = shallow(<CardForm />);
  });

  it('should render the <CardForm /> Component correctly', () => {   
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an email input tag', () => {
    expect(wrapper.find('input[name="givenname"]').exists()).toBe(true);
  });

  it('should render an email input tag', () => {
    expect(wrapper.find('input[name="surname"]').exists()).toBe(true);
  });

  it('should render an email input tag', () => {
    expect(wrapper.find('input[name="email"]').exists()).toBe(true);
  });

  it('should render an email input tag', () => {
    expect(wrapper.find('input[name="phone"]').exists()).toBe(true);
  });

  it('should render an email input tag', () => {
    expect(wrapper.find('input[name="houseno"]').exists()).toBe(true);
  });

  it('should render an email input tag', () => {
    expect(wrapper.find('input[name="street"]').exists()).toBe(true);
  });

  it('should render an email input tag', () => {
    expect(wrapper.find('input[name="suburb"]').exists()).toBe(true);
  });

  it('should render an email input tag', () => {
    expect(wrapper.find('input[name="state"]').exists()).toBe(true);
  });

  it('should render an email input tag', () => {
    expect(wrapper.find('input[name="postcode"]').exists()).toBe(true);
  });

  it('should render an email input tag', () => {
    expect(wrapper.find('input[name="country"]').exists()).toBe(true);
  });


  it('should render a upload button', () => {
    expect(wrapper.find('button[name="upload"]').exists()).toBe(true);
  });

  it('should render a submit button', () => {
    expect(wrapper.find('button[type="submit"]').exists()).toBe(true);
  });

  it('on submitting, a submit handler function should be triggered on click event', () => {
    const fn = jest.fn();
    const wrapper = shallow(<CardForm onSubmit={fn} />);
    wrapper.find('form').simulate('submit');
    wrapper.find('button[name="submit"]').simulate('submit');
  });
});
