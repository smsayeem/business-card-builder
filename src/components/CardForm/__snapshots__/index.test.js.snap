// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should render the <CardForm /> Component correctly 1`] = `
<div
  className="form-container"
>
  <header>
    <h1>
      hCard Builder
    </h1>
  </header>
  <form
    onSubmit={[Function]}
  >
    <div
      className="divider"
    >
      <p>
        personal details
      </p>
    </div>
    <section
      className="form-center"
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="givenname"
        >
          given name
        </label>
        <input
          className="form-input"
          id="givenname"
          name="givenname"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="surname"
        >
          surname
        </label>
        <input
          className="form-input"
          id="surname"
          name="surname"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="email"
        >
          email
        </label>
        <input
          className="form-input"
          id="email"
          name="email"
          onChange={[Function]}
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="phone"
        >
          phone
        </label>
        <input
          className="form-input"
          id="phone"
          name="phone"
          onChange={[Function]}
          type="text"
        />
      </div>
    </section>
    <div
      className="divider"
    >
      <p>
        personal details
      </p>
    </div>
    <section
      className="form-center"
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="houseno"
        >
          house name or #
        </label>
        <input
          className="form-input"
          id="houseno"
          name="houseno"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="street"
        >
          street
        </label>
        <input
          className="form-input"
          id="street"
          name="street"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="suburb"
        >
          suburb
        </label>
        <input
          className="form-input"
          id="suburb"
          name="suburb"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="state"
        >
          state
        </label>
        <input
          className="form-input"
          id="state"
          name="state"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="postcode"
        >
          post code
        </label>
        <input
          className="form-input"
          id="postcode"
          name="postcode"
          onChange={[Function]}
          type="number"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="country"
        >
          country
        </label>
        <input
          className="form-input"
          id="country"
          name="country"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="btn-group "
      >
        <input
          hidden={true}
          id="upload-button"
          type="file"
        />
        <button
          className="btn btn-avatar"
          name="upload"
          onClick={[Function]}
        >
          Upload Avatar
        </button>
        <button
          className="btn btn-hcard"
          name="submit"
          onClick={[Function]}
          type="submit"
        >
          Create hCard
        </button>
      </div>
    </section>
  </form>
</div>
`;
